src/components/Input.react.js

```diff
@@ -89,6 +92,11 @@ export default class Input extends PureComponent {
         );
     }
 
+    getValueAsNumber(value) {
+        const numericValue = convert(value);
+        return numericValue % 1 === 0 ? Math.floor(numericValue) : numericValue;
```

Comment:
@shammamah Can you explain the reasoning for this line?

Reply:
I wasn't sure if it would automatically convert to int vs. remaining as a float. Fixed in 24b3b9d.